src/BSON/DBPointer.c

```diff
@@ -46,7 +46,7 @@ static bool php_phongo_dbpointer_init(php_phongo_dbpointer_t* intern, const char
 	intern->ref     = estrndup(ref, ref_len);
 	intern->ref_len = ref_len;
 
-	strncpy(intern->id, id, sizeof(intern->id));
+	strncpy(intern->id, id, sizeof(intern->id) - 1);
```

Comment:
This looks like a bug fix, even though we might never hit this thanks to the `bson_oid_is_valid` check above. Should it have a separate ticket, though?

Reply:
If you want to include this in the changelog for tracking purposes, I'm happy to file a ticket for this 👍

Reply:
https://jira.mongodb.org/browse/PHPC-1775

Reply:
I'll extract this in a separate PR that includes the other sanity checks you mentioned in the ticket.

Reply:
Did you mean to pull this commit out? Looks like it's still here.